cuqi/density/_density.py

```diff
@@ -30,6 +31,18 @@ def __init__(self, name: Optional[str] = None):
         self.name = name
         self._constant = 0 # Precomputed constant to add to the log probability.
 
+    @property
+    def name(self):
+        """ Name of the random variable associated with the density. """
+        if self._name is None: # If none extract the name from the stack
+            name = cuqi.utilities.get_python_variable_name(self)
```

Comment:
if the name is None after search, may be we we can print a warning message to suggest that the user can specify the distribution name if needed.

Reply:
Good suggestion. There is a warning message in the python variable name function. I updated it a bit.